js/core/component.js

```diff
@@ -383,21 +351,21 @@ var Component = Class.inherit({
     },
 
     initialOption: function(optionName) {
-        var currentOptions,
-            currentInitialized = this._initialized;
         if(!this._initialOptions) {
-            currentOptions = this._options;
-            this._options = {};
-            this._initialized = false;
-            this._setDefaultOptions();
-            this._setOptionsByDevice(currentOptions.defaultOptionsRules);
-
-            this._initialOptions = this._options;
-            this._options = currentOptions;
-            this._initialized = currentInitialized;
+            this._initialOptions = {};
+            this._initialOptions = this._getDefaultOptions();
+            const rulesOptions = this._getOptionByRules(this._getOptionSilent("defaultOptionsRules"));
```

Comment:
Why do you use `this._getOptionSilent` and `_optionManager._setSilent` at the same time. It looks like component's methods should be used everywhere